### PR TITLE
[TBAA] Tests for invalid tbaa.struct metadata

### PR DESCRIPTION
**Pull request by juliannagele**
These tests show invalid tbaa.struct metadata that is currently accepted in preparation for a change to the IR Verifier that will then reject it.

### CODE REVIEW

**Comment by juliannagele** [comment]
I don't think this got labelled, so adding (hopefully the right) people manually: @jdoerfert and @dobbelaj-snps

**Review by dobbelaj-snps — APPROVED** [review]
lgtm

File: llvm/test/Verifier/tbaa-struct.ll [hunk 1]
```diff
@@ -0,0 +1,38 @@
+; RUN: llvm-as < %s 2>&1
+
+define void @test_overlapping_regions(ptr %a1) {
```
Comment (fhahn) [inline llvm/test/Verifier/tbaa-struct.ll:3]: might be good to add a FIXME that the verifier should reject the invalid `!tbaa.struct` nodes.

Comment (juliannagele) [reply llvm/test/Verifier/tbaa-struct.ll:5]: Thanks, done!

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**